app/Actions/Auth/RegisterUserAction.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Hedonist\Actions\Auth;
+
+
+use Hedonist\Actions\ActionInterface;
+use Hedonist\Actions\Auth\Responses\RegisterResponse;
+use Hedonist\Actions\RequestInterface;
+use Hedonist\Actions\ResponseInterface;
+use Hedonist\Entities\User;
+use Hedonist\Exceptions\Auth\EmailAlreadyExistsException;
+use Hedonist\Repositories\User\UserRepositoryInterface;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Support\Facades\Hash;
+
+class RegisterUserAction implements ActionInterface
+{
+    private $repository;
+
+    public function __construct(UserRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function execute(RequestInterface $request):ResponseInterface
+    {
+        $user = new User();
+        if($this->repository->getByEmail($request->getEmail()) !== null){
+            throw new EmailAlreadyExistsException();
+        }
+        $user->email = $request->getEmail();
+        $user->password = Hash::make($request->getPassword());
+        $user->name = $request->getName();
+        $this->repository->save($user);
+        event(new Registered($user));
```

Comment:
Try to inject Illuminate\Event\Dispather into constructor and then use `$this->dispacher->dispatch(new Registered(...));`

Reply:
used Event::dispatch() instead

Reply:
@klesogor `Event::dispatch()` also might be. But you will have troubles while testing this RegisterUserAction class, because `Event` facade is not explicit dependency and injected statically. DI here will save your time in future.